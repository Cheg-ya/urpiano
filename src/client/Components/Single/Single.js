import React, { Component } from 'react';
import './Single.css';

class Single extends Component {
  render() {
    return (
      <div className="singleContainer">
        <div className="singleCover">
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default Single;
