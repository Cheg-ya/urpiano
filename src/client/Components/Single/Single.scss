.singleCover {
  position: relative;

  .imageBackgroundCover {
    width: 100%;
    height: 98vh;

    .backgroundImage {
      width: 100%;
      height: 100%;
    }
  }
    
  .pianoCover {
    position: absolute;
    width: 80%;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
  }
}
