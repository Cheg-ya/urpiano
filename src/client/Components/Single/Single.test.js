import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Single from './Single';

describe('Single component test', () => {
  it('render correctly', () => {
    const component = shallow(<Single />);
    const tree = renderer.create(<Single />).toJSON();

    expect(component.exists()).toBe(true);
    expect(component.state()).toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('renders its children DOM node', () => {
    const tree = renderer.create(
      <Single>
        <div className="target">Hello</div>
      </Single>
    ).toJSON();

    const component = shallow((
      <Single>
        <div className="target">Hello</div>
      </Single>
    ));

    expect(component.contains(<div className="target">Hello</div>)).toBe(true);
    expect(tree).toMatchSnapshot();
  });
});
