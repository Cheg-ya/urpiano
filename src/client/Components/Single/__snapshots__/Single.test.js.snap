// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single component test render correctly 1`] = `
<div
  className="singleContainer"
>
  <div
    className="singleCover"
  >
    <div
      className="imageBackgroundCover"
    >
      <img
        alt=""
        className="backgroundImage"
        src="https://images.unsplash.com/photo-1552186118-22d86b3559b7?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=2500&fit=max&ixid=eyJhcHBfaWQiOjcwNjZ9"
      />
    </div>
    <div
      className="pianoCover"
    />
  </div>
</div>
`;

exports[`Single component test renders its children DOM node 1`] = `
<div
  className="singleContainer"
>
  <div
    className="singleCover"
  >
    <div
      className="imageBackgroundCover"
    >
      <img
        alt=""
        className="backgroundImage"
        src="https://images.unsplash.com/photo-1552186118-22d86b3559b7?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=2500&fit=max&ixid=eyJhcHBfaWQiOjcwNjZ9"
      />
    </div>
    <div
      className="pianoCover"
    >
      <div
        className="target"
      >
        Hello
      </div>
    </div>
  </div>
</div>
`;
